Foundation/FileHandle.swift

```diff
@@ -345,7 +346,19 @@ extension FileHandle {
     }
 
     open func waitForDataInBackgroundAndNotify() {
-        NSUnimplemented()
+        let global = DispatchQueue.global(qos: .background)
+        let channel = DispatchIO(type: .stream, fileDescriptor: fileDescriptor, queue: global) { _ in }
+
+        // The desire is to trigger the handler blocka as soon as there is any data in the channel.
+        channel.setLimit(lowWater: 1)
+
+        channel.read(offset: 0, length: Int.max, queue: global) { (_, _, error) in
```

Comment:
- this reads all the data for no reason
- also if this is a socket/pipe this is a permanent leak of a file descriptor
 - you could just use a `DispatchSource.makeReadSource` and get the notifications but not the reading. As soon as you get the first notification you then want to `.cancel` the `DispatchSource` as you only want to get notified once.